Sources/SemanticIndex/UpdateIndexStoreTaskDescription.swift

```diff
@@ -264,72 +276,150 @@ package struct UpdateIndexStoreTaskDescription: IndexTaskDescription {
       )
     }
 
+    // Compute the partitions within which we can perform multi-file indexing. For this we gather all files tha may only
+    // be indexed by itself in `partitions` an collect all other files in `fileInfosByBuildSettings` so that we can
+    // create a partition for all files that share the same build settings.
+    // In most cases, we will only end up with a single partition for each `UpdateIndexStoreTaskDescription` since
+    // `UpdateIndexStoreTaskDescription.batches(toIndex:)` tries to batch the files in a way such that all files within
+    // the batch can be indexed by a single compiler invocation. However, we might discover that two Swift files within
+    // the same target have different build settings in the build server. In that case, the best thing we can do is
+    // trigger two compiler invocations.
```

Comment:
How would this happen 🤔? For C-family, sure. But I would expect all files within a target to have the same build settings for Swift.

It's super unfortunate we have to look through the arguments to find `-index-unit-output-path` for every settings per file, even though they should all be the same for every batch (except for the output path).

Reply:
I've seen extremely large compiler arguments, it all adds up 🤷 But I'm fine leaving in handling different arguments, can always re-visit if we notice it being a real issue.